### PR TITLE
fault handler attempt

### PR DESCRIPTION
**Pull request by Rand-Sai**
I modified the "fault.c" file, attempted to use a if loop to pop faults off the queue and made a switch statement to create priorities for tasks.

### CODE REVIEW

File: Core/Src/fault.c [hunk 1]
```diff
@@ -1,3 +1,45 @@
 #include "fault.h"
 
 osMessageQueueId_t fault_handle_queue;
+
+//Function to determine priority base on severity
+void priority(void) {
```
Comment (nwdepatie) [inline Core/Src/fault.c:6]: So this should be the main "task" which we will assign a priority and tell the kernel to schedule on boot. For references on other tasks, check the functions that start with `vSomething` in monitor.c. These have a thread attribute struct associated with them that we use to initialize in main.c

File: Core/Src/fault.c [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+    osThreadId_t id;
+    MSGQUEUE_OBJ_t msg;
```
Comment (nwdepatie) [inline Core/Src/fault.c:8]: So the type of the message that we receive depends on what we are actually queuing, in this case I believe the type is `FaultStatus_t` or something I forgot exactly what it is

File: Core/Src/fault.c [hunk 3]
```diff
@@ -1,3 +1,45 @@
...
+    osStatus_t status;
+    // Wait until a message is in the queue, send messages when they are in the queue
+	for(;;) {
+        status = osMessageQueueGet(fault_handle_queue, &msg, NULL, 0U);   // wait for message
+        id = osThreadGetId(); // Obtain ID of current running thread
```
Comment (nwdepatie) [inline Core/Src/fault.c:13]: You shouldn't need to get the ID each time

File: Core/Src/fault.c [hunk 4]
```diff
@@ -1,3 +1,45 @@
...
+        if (status == osOK) {
+        
+        // process data
+        switch (fault_sev_t) {
```
Comment (nwdepatie) [inline Core/Src/fault.c:17]: You want to access the severity field of the `fault_t` or whatever data type you get from the queue. The idea is that we are popping off a fault and then routing to a case for the severity

File: Core/Src/fault.c [hunk 5]
```diff
@@ -1,3 +1,45 @@
...
+        case 'DEFCON1': //Higest(1st) Priority
+            osThreadSetPriority(id, osPriorityHigh);
```
Comment (nwdepatie) [inline Core/Src/fault.c:19]: I don't think we need to set the priority here, tbh I would just leave this blank for now apart from the `break;`'s and we will handle faults in a later ticket

**Review by nwdepatie — CHANGES_REQUESTED** [review]
Okay so there is def a bit of work to be done, but solid first attempt. I think I've sent a few reference commits but the typical flow of adding a task is:
1. create a function with similar parameters to the other `vSomething` tasks in other files (particularly monitor.c)
2. create a threadattributes struct, similar to the stuff implemented in monitor.h. These specify the attributes of each task
3. Actually initialize the task in main.c, there is a bunch of functions grouped together in main that are all initializing the task

**Comment by Rand-Sai** [comment]
Made edits based on suggestions provided. Used functions from "https://www.keil.com/pack/doc/cmsis/RTOS2/html/group__CMSIS__RTOS__Message.html" as references.

**Review by nwdepatie — APPROVED** [review]
This looks like a solid start

**Pull request merged**